accounting/src/delta/tests.rs

```diff
@@ -101,22 +107,22 @@ fn data_delta_delta(
 
 #[rstest]
 #[rustfmt::skip]
-#[case(None,      DataDelta::Create('a'),      DataDelta::Modify('a', 'b'),  DataDelta::Delete('b'),      None)]
-#[case(None,      DataDelta::Create('a'),      DataDelta::Modify('a', 'b'),  DataDelta::Modify('b', 'c'), Some('c'))]
-#[case(None,      DataDelta::Create('a'),      DataDelta::Delete('a'),       DataDelta::Create('a'),      Some('a'))]
-#[case(Some('a'), DataDelta::Delete('a'),      DataDelta::Create('a'),       DataDelta::Delete('a'),      None)]
-#[case(Some('a'), DataDelta::Delete('a'),      DataDelta::Create('a'),       DataDelta::Modify('a', 'b'), Some('b'))]
-#[case(Some('a'), DataDelta::Modify('a', 'b'), DataDelta::Delete('a'),       DataDelta::Create('a'),      Some('b'))]
-#[case(Some('a'), DataDelta::Modify('a', 'b'), DataDelta::Modify('b', 'c'),  DataDelta::Delete('c'),      None)]
-#[case(Some('a'), DataDelta::Modify('a', 'b'), DataDelta::Modify('b', 'c'),  DataDelta::Modify('c', 'd'), Some('d'))]
+#[case(None,      DataDelta::new(None, Some('a')),      DataDelta::new(Some('a'), Some('b')),  DataDelta::new(Some('b'), None),      None)]
+#[case(None,      DataDelta::new(None, Some('a')),      DataDelta::new(Some('a'), Some('b')),  DataDelta::new(Some('b'), Some('c')), Some('c'))]
+#[case(None,      DataDelta::new(None, Some('a')),      DataDelta::new(Some('a'), None),       DataDelta::new(None, Some('a')),      Some('a'))]
+#[case(Some('a'), DataDelta::new(Some('a'), None),      DataDelta::new(None, Some('a')),       DataDelta::new(Some('a'), None),      None)]
+#[case(Some('a'), DataDelta::new(Some('a'), None),      DataDelta::new(None, Some('a')),       DataDelta::new(Some('a'), Some('b')), Some('b'))]
+#[case(Some('a'), DataDelta::new(Some('a'), Some('b')), DataDelta::new(Some('b'), None),       DataDelta::new(None, Some('a')),      Some('a'))]
+#[case(Some('a'), DataDelta::new(Some('a'), Some('b')), DataDelta::new(Some('b'), Some('c')),  DataDelta::new(Some('c'), None),      None)]
+#[case(Some('a'), DataDelta::new(Some('a'), Some('b')), DataDelta::new(Some('b'), Some('c')),  DataDelta::new(Some('c'), Some('d')), Some('d'))]
```

Comment:
You can add some helper constructors to streamline this a bit:

* `DataDelta::modify('a', 'b')` => `DataDelta::new(Some('a'), Some('b'))`
* `DataDelta::create('c')` => `DataDelta::new(None, Some('b'))`
* etc...

Maybe it should be called `new_modify` instead of just `modify` (similar with the rest), not a strong opinion on that either way